riscv.c

```diff
@@ -758,47 +758,122 @@ static bool op_amo(struct riscv_t *rv, uint32_t inst)
     return true;
 }
 #else
-#define op_amo NULL
+#define op_amo OP_UNIMP
 #endif  // ENABLE_RV32A
 
 /* No RV32F support */
-#define op_load_fp NULL
-#define op_store_fp NULL
-#define op_fp NULL
-#define op_madd NULL
-#define op_msub NULL
-#define op_nmsub NULL
-#define op_nmadd NULL
+#define op_load_fp OP_UNIMP
+#define op_store_fp OP_UNIMP
+#define op_fp OP_UNIMP
+#define op_madd OP_UNIMP
+#define op_msub OP_UNIMP
+#define op_nmsub OP_UNIMP
+#define op_nmadd OP_UNIMP
+
+// handler for all unimplemented opcodes
+static bool op_unimp(struct riscv_t *rv, uint32_t inst UNUSED)
+{
+    rv_except_illegal_inst(rv);
+    return false;
+}
 
 // opcode handler type
 typedef bool (*opcode_t)(struct riscv_t *rv, uint32_t inst);
 
-// clang-format off
-// opcode dispatch table
-static const opcode_t opcodes[] = {
-//  000        001          010       011          100        101       110   111
-    op_load,   op_load_fp,  NULL,     op_misc_mem, op_op_imm, op_auipc, NULL, NULL, // 00
-    op_store,  op_store_fp, NULL,     op_amo,      op_op,     op_lui,   NULL, NULL, // 01
-    op_madd,   op_msub,     op_nmsub, op_nmadd,    op_fp,     NULL,     NULL, NULL, // 10
-    op_branch, op_jalr,     NULL,     op_jal,      op_system, NULL,     NULL, NULL, // 11
-};
-// clang-format on
-
 void rv_step(struct riscv_t *rv, int32_t cycles)
 {
     assert(rv);
     const uint64_t cycles_target = rv->csr_cycle + cycles;
+    uint32_t inst, index;
+// clang-format off
```

Comment:
Let's remove the `clang-format` mark. Stick on the formatting.
```cpp

#define OP_UNIMP op_unimp                                                                                                                                     
#ifdef ENABLE_COMPUTED_GOTO
#define OP(instr) &&op_##instr
#define TABLE_TYPE const void *
#else
#define OP(instr) op_##instr
#define TABLE_TYPE const opcode_t
#endif
```

The above is acceptable and common.

Reply:
Alternatively, you can add some visual annotation as following:
```c
#ifdef ENABLE_COMPUTED_GOTO
#define OP(instr) &&op_##instr
#define TABLE_TYPE const void *
#else /* without computed goto */
#define OP(instr) op_##instr
#define TABLE_TYPE const opcode_t
#endif
```